### PR TITLE
Hardcoded dendrites (and initial attempt at learning dendrites)

### PR DESCRIPTION
**Pull request by karangrewal**
These project files run a dendritic network in a continual learning scenario where each task consists of 2-way classification between samples drawn from 2 distinct multivariate normal distributions. The dendritic network can be run either with hardcoded dendrites (which remain fixed) or it can learn dendritic weights.

The file `run_dendritic_network.py` can be executed, and utilizes Vernon experiment classes to run the continual learning experiment.

Here's how the various network parameters can be adjusted: in `model_args` (in the config), set
- `kw` --> set to `True` to include k-Winners with 5% on after each dendritic layer, `False` for no k-Winners
- `weight_init` --> must be either `"kaiming"` for regular Kaiming init, or `"modified"` for modified Kaiming init that takes sparsity into account,
- `dendrite_init` --> can be `"kaiming"` or `"modified"` just as above, or `"hardcoded"` to invoke non-overlapping subnetworks and keep fixed.

### CODE REVIEW

File: projects/dendrites/gaussian_classification/run_dendritic_network.py [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+# ------ Experiment class
+class DendritesExperiment(mixins.RezeroWeights,
+                          ContinualLearningExperiment):
+    pass
+
+
+# ------ Network
+class DendriticMLP(nn.Module):
+    """
+    Dendrite segments receive context directly as input.
```
Comment (subutai) [inline projects/dendrites/gaussian_classification/run_dendritic_network.py:70]: Nice ASCII diagram 😄

File: projects/dendrites/gaussian_classification/run_dendritic_network.py [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+                      |
+                  _________
+    context -->  |_________|  # first linear layer with dendrites
+                      ^
+                      |
+                    input
+    """
+
+    def __init__(self, input_size, output_size, hidden_size, num_segments, dim_context,
+                 weight_init, dendrite_init, kw, duty_cycle_period=15000,
```
Comment (subutai) [inline projects/dendrites/gaussian_classification/run_dendritic_network.py:88]: In the batch configurations we'd want `duty_cycle_period` to be the default, not 15,000

**Review by subutai — APPROVED** [review]

**Comment by subutai** [comment]
You'll want to merge your branch with the latest from master before merging the PR.

File: projects/dendrites/gaussian_classification/run_dendritic_network.py (hunk 2 above)
Comment (karangrewal) [reply projects/dendrites/gaussian_classification/run_dendritic_network.py]: Fixed.

**Pull request merged**